data-model/src/test/java/org/pdxfinder/graph/dao/TreatmentProtocolTest.java

```diff
@@ -0,0 +1,302 @@
+
+        TreatmentProtocol treatmentProtocol = new TreatmentProtocol();
+        List<TreatmentComponent> componentsList= new ArrayList<>();
+
+        TreatmentComponent controlTC = new TreatmentComponent();
+        TreatmentComponent drug1 = new TreatmentComponent();
+        TreatmentComponent drug2 = new TreatmentComponent();
+
+        controlTC.setDose(CONTROL_DOSE_1);
+        drug1.setDose(DRUG_DOSE_2);
+        drug2.setDose(DRUG_DOSE_3);
+
+        controlTC.setDuration(CONTROL_DURATION_1);
+        drug1.setDuration(DRUG_DURATION_2);
+        drug2.setDuration(DRUG_DURATION_3);
+
+        controlTC.setTreatment(treatments.get(0));
+        controlTC.setType("Control");
+
+        drug1.setTreatment(treatments.get(1));
+        drug2.setTreatment(treatments.get(2));
+
+        componentsList.add(controlTC);
+        componentsList.add(drug1);
+        componentsList.add(drug2);
+
+        Response testResponse = new Response(RESPONSE_STRING, treatmentProtocol);
+
+        treatmentProtocol.setComponents(componentsList);
+        treatmentProtocol.setResponse(testResponse);
+
+        treatmentProtocolList.add(treatmentProtocol);
+    }
+
+    private void createMockDatasets2() {
+
+        Treatment treatment;
+        treatment = createTreatmentsAndOntologies2();
+        createTreatmentComponentsAndProtocols2(treatment);
+    }
+
+    private Treatment createTreatmentsAndOntologies2() {
+
+        Treatment treatment1 = new Treatment(FIRST_TREATMENT_NAME);
+        TreatmentToOntologyRelationship ontologyRelationship1 = new TreatmentToOntologyRelationship();
+        treatment1.setTreatmentToOntologyRelationship(ontologyRelationship1);
+
+        return treatment1;
+    }
+
+    private void createTreatmentComponentsAndProtocols2(Treatment treatments){
+
+        TreatmentProtocol treatmentProtocol = new TreatmentProtocol();
+        TreatmentProtocol treatmentProtocol2 = new TreatmentProtocol();
+
+        List<TreatmentComponent> componentsList1 = new ArrayList<>();
+        List<TreatmentComponent> componentsList2 = new ArrayList<>();
+
+        TreatmentComponent controlTC = new TreatmentComponent();
+        TreatmentComponent drug1 = new TreatmentComponent();
+
+        controlTC.setTreatment(treatments);
+        controlTC.setType("Control");
+
+        controlTC.setTreatment(treatments);
+
+        componentsList1.add(controlTC);
+        componentsList2.add(drug1);
+
+        Response testResponse = new Response(RESPONSE_STRING, treatmentProtocol);
+        Response testResponse2 = new Response(RESPONSE_STRING, treatmentProtocol2);
+
+        treatmentProtocol.setComponents(componentsList1);
+        treatmentProtocol2.setComponents(componentsList2);
+
+        treatmentProtocol.setResponse(testResponse);
+        treatmentProtocol2.setResponse(testResponse2);
+
+        treatmentProtocolList.add(treatmentProtocol);
+        treatmentProtocolList.add(treatmentProtocol2);
```

Comment:
Try for side effect free methods (encapsulate all actions that change the state of the class).  Better to return a list of TreatmentProtocols and add them to the class level variable in the caller.

Reply:
Thanks for the feedback Jeremy. You can be first mate on the USS AlexIsNumber1!!